pkg/infrabin/http_test.go

```diff
@@ -297,24 +300,25 @@ func TestProxyHandler(t *testing.T) {
 }
 
 func TestAWSHandler(t *testing.T) {
+
 	mockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusOK)
 		if _, err := w.Write([]byte("{}")); err != nil {
 			t.Fatalf("Failed to write fake response body: %v", err)
 		}
 	}))
-	config := &Config{
-		EnableProxyEndpoint: true,
-		AWSMetadataEndpoint: mockServer.URL,
-	}
-	is := &InfrabinService{Config: config}
+
+	viper.Set("proxyEndpoint", true)
+	viper.Set("awsMetadataEndpoint", mockServer.URL)
+
+	is := &InfrabinService{}
```

Comment:
Can be removed too.
```suggestion
```

Reply:
Ok, so I didn't realise I can commit your recommended changes - but I've implemented them all as described and added some basic unit tests to capture the default configuration behaviour so that there is no inadvertent change made.